gnovm/pkg/gnolang/values_conversions.go

```diff
@@ -1253,7 +1257,10 @@ func ConvertUntypedBigdecTo(dst *TypedValue, bv BigdecValue, t Type) {
 	case Float64Kind:
 		dst.T = t
 		dst.V = nil
-		f64, _ := bd.Float64()
+		f64, err := bd.Float64()
+		if err != nil {
+			panic(fmt.Errorf("cannot convert untyped bigdec to float64: %w", err))
```

Comment:
When is this situation possible?

Reply:
Every error is possible and must be handled.

Reply:
>Every error is possible and must be handled

```
rg "panic.*should not happen" | wc -l
322
```

:trollface: 

Seriously, though, this situation should only happen if `strconv.ParseFloat` returns an error after parsing the string representation of the decimal. The only case where it can happen seems to be this one:

    If s is syntactically well-formed but is more than 1/2 ULP away from the
    largest floating point number of the given size, ParseFloat returns f =
    ±Inf, err.Err = ErrRange.

So, intrinsically it is the same as the below `math.IsInf`. Though I'm good to keep this :+1:

Reply:
>  ```
> rg "panic.*should not happen" | wc -l
> 322
>	```

I reassure you that with enough code-running time, all these "should not happen" errors will be thrown to someone's face :P